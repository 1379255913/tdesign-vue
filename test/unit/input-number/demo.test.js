import { mount } from '@vue/test-utils';
import center from '@/examples/input-number/demos/center.vue';
import defaultDemo from '@/examples/input-number/demos/default.vue';
import disabled from '@/examples/input-number/demos/disabled.vue';
import formatter from '@/examples/input-number/demos/formatter.vue';
import left from '@/examples/input-number/demos/left.vue';
import size from '@/examples/input-number/demos/size.vue';
import step from '@/examples/input-number/demos/step.vue';
import undefinedDemo from '@/examples/input-number/demos/empty.vue';

describe('InputNumber', () => {
  it('center demo works fine', () => {
    const wrapper = mount(center);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('defaultDemo demo works fine', () => {
    const wrapper = mount(defaultDemo);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('disabled demo works fine', () => {
    const wrapper = mount(disabled);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('formatter demo works fine', () => {
    const wrapper = mount(formatter);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('left demo works fine', () => {
    const wrapper = mount(left);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('size demo works fine', () => {
    const wrapper = mount(size);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('step demo works fine', () => {
    const wrapper = mount(step);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('undefined demo works fine', () => {
    const wrapper = mount(undefinedDemo);
    expect(wrapper.element).toMatchSnapshot();
  });
});
