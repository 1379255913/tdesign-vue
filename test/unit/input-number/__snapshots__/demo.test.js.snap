// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputNumber center demo works fine 1`] = `
<div>
  <div
    class="t-input-number t-size-m"
  >
    <span
      class="t-input-number__decrease"
    >
      <svg
        class="t-icon t-icon-remove t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M224 470.39872H800V553.59872H224V470.39872Z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner t-input-number-text-align"
        placeholder=""
        ref="refInputElem"
      />
    </div>
    <div
      class="t-input-number__increase"
    >
      <svg
        class="t-icon t-icon-add t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M470.39872 553.59872V800H553.59872V553.59872H800V470.39872H553.59872V224H470.39872V470.39872H224V553.59872H470.39872Z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`InputNumber defaultDemo demo works fine 1`] = `
<div>
  <div
    class="t-input-number t-size-m"
  >
    <span
      class="t-input-number__decrease"
    >
      <svg
        class="t-icon t-icon-remove t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M224 470.39872H800V553.59872H224V470.39872Z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner t-input-number-text-align"
        placeholder=""
        ref="refInputElem"
      />
    </div>
    <div
      class="t-input-number__increase"
    >
      <svg
        class="t-icon t-icon-add t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M470.39872 553.59872V800H553.59872V553.59872H800V470.39872H553.59872V224H470.39872V470.39872H224V553.59872H470.39872Z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`InputNumber disabled demo works fine 1`] = `
<div>
  <div
    class="t-input-number t-size-m t-is-disabled"
  >
    <span
      class="t-input-number__decrease t-is-disabled"
    >
      <svg
        class="t-icon t-icon-remove t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M224 470.39872H800V553.59872H224V470.39872Z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner t-is-disabled t-input-number-text-align"
        disabled="disabled"
        placeholder=""
        ref="refInputElem"
      />
    </div>
    <div
      class="t-input-number__increase t-is-disabled"
    >
      <svg
        class="t-icon t-icon-add t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M470.39872 553.59872V800H553.59872V553.59872H800V470.39872H553.59872V224H470.39872V470.39872H224V553.59872H470.39872Z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`InputNumber format demo works fine 1`] = `
<div>
  <div
    class="t-input-number t-size-m"
  >
    <span
      class="t-input-number__decrease"
    >
      <svg
        class="t-icon t-icon-remove t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M224 470.39872H800V553.59872H224V470.39872Z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner t-input-number-text-align"
        placeholder=""
        ref="refInputElem"
      />
    </div>
    <div
      class="t-input-number__increase"
    >
      <svg
        class="t-icon t-icon-add t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M470.39872 553.59872V800H553.59872V553.59872H800V470.39872H553.59872V224H470.39872V470.39872H224V553.59872H470.39872Z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`InputNumber left demo works fine 1`] = `
<div>
  <div
    class="t-input-number t-size-m t-is-controls-right"
  >
    <span
      class="t-input-number__decrease"
    >
      <svg
        class="t-icon t-icon-chevron-down t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M226.57792 413.41696L285.40928 354.58624 511.9936 581.17056 738.5792 354.58624 797.408 413.41696 511.9936 698.8352 226.57792 413.41696Z"
          fill-opacity=".9"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner"
        placeholder=""
        ref="refInputElem"
      />
    </div>
    <div
      class="t-input-number__increase"
    >
      <svg
        class="t-icon t-icon-chevron-up t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M797.408 610.60864L738.5792 669.44 511.9936 442.85568 285.40928 669.44 226.57792 610.60864 511.9936 325.19296 797.408 610.60864Z"
          fill-opacity=".9"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`InputNumber size demo works fine 1`] = `
<div>
  <div
    class="t-input-number t-size-s"
  >
    <span
      class="t-input-number__decrease"
    >
      <svg
        class="t-icon t-icon-remove t-size-s"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M224 470.39872H800V553.59872H224V470.39872Z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner t-input-number-text-align"
        placeholder=""
        ref="refInputElem"
      />
    </div>
    <div
      class="t-input-number__increase"
    >
      <svg
        class="t-icon t-icon-add t-size-s"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M470.39872 553.59872V800H553.59872V553.59872H800V470.39872H553.59872V224H470.39872V470.39872H224V553.59872H470.39872Z"
        />
      </svg>
    </div>
  </div>
   
  <div
    class="t-input-number t-size-m"
    style="margin: 0px 50px;"
  >
    <span
      class="t-input-number__decrease"
    >
      <svg
        class="t-icon t-icon-remove t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M224 470.39872H800V553.59872H224V470.39872Z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner t-input-number-text-align"
        placeholder=""
        ref="refInputElem"
      />
    </div>
    <div
      class="t-input-number__increase"
    >
      <svg
        class="t-icon t-icon-add t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M470.39872 553.59872V800H553.59872V553.59872H800V470.39872H553.59872V224H470.39872V470.39872H224V553.59872H470.39872Z"
        />
      </svg>
    </div>
  </div>
   
  <div
    class="t-input-number t-size-l"
  >
    <span
      class="t-input-number__decrease"
    >
      <svg
        class="t-icon t-icon-remove t-size-l"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M224 470.39872H800V553.59872H224V470.39872Z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner t-input-number-text-align"
        placeholder=""
        ref="refInputElem"
      />
    </div>
    <div
      class="t-input-number__increase"
    >
      <svg
        class="t-icon t-icon-add t-size-l"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M470.39872 553.59872V800H553.59872V553.59872H800V470.39872H553.59872V224H470.39872V470.39872H224V553.59872H470.39872Z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`InputNumber step demo works fine 1`] = `
<div>
  <div
    class="t-input-number t-size-m"
  >
    <span
      class="t-input-number__decrease"
    >
      <svg
        class="t-icon t-icon-remove t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M224 470.39872H800V553.59872H224V470.39872Z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner t-input-number-text-align"
        placeholder=""
        ref="refInputElem"
      />
    </div>
    <div
      class="t-input-number__increase"
    >
      <svg
        class="t-icon t-icon-add t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M470.39872 553.59872V800H553.59872V553.59872H800V470.39872H553.59872V224H470.39872V470.39872H224V553.59872H470.39872Z"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`InputNumber undefined demo works fine 1`] = `
<div>
  <div
    class="t-input-number t-size-m"
  >
    <span
      class="t-input-number__decrease"
    >
      <svg
        class="t-icon t-icon-remove t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M224 470.39872H800V553.59872H224V470.39872Z"
          fill-rule="evenodd"
        />
      </svg>
    </span>
    <div
      class="t-input"
    >
      <input
        autocomplete="off"
        class="t-input__inner t-input-number-text-align"
        placeholder="输入"
        ref="refInputElem"
      />
    </div>
    <div
      class="t-input-number__increase"
    >
      <svg
        class="t-icon t-icon-add t-size-m"
        version="1.1"
        viewBox="0 0 1024 1024"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M470.39872 553.59872V800H553.59872V553.59872H800V470.39872H553.59872V224H470.39872V470.39872H224V553.59872H470.39872Z"
        />
      </svg>
    </div>
  </div>
</div>
`;
